### PR TITLE
Add compute instance creation for provisioning of managed vnet

### PR DESCRIPTION
**Pull request by oloomi**
## Description

This PR includes the following changes:
- Creates an AML compute instance (with no public IP) after the workspace is created, to trigger provisioning of managed vnet
- Upgrades API version from `2024-04-01-preview` to `2024-04-01`

## Type of Change

- [ ] Non-module change (e.g. CI/CD, documentation, etc.)
- [ ] Azure Verified Module updates:
  - [ ] Bugfix containing backwards compatible bug fixes
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [x] Feature update backwards compatible feature updates.
  - [ ] Breaking changes.
  - [ ] Update to documentation

# Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] I did run all  [pre-commit](https://azure.github.io/Azure-Verified-Modules/contributing/terraform/terraform-contribution-flow/#5-run-pre-commit-checks) checks

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -74,6 +74,31 @@ resource "azapi_resource" "aiserviceconnection" {
...
+  count = var.create_compute_instance ? 1 : 0
+
+  type = "Microsoft.MachineLearningServices/workspaces/computes@2024-04-01"
+  body = jsonencode({
+    properties = {
+      computeType = "ComputeInstance"
+      properties = {
+        enableNodePublicIp = false
+        # enableSSO = bool
+        # idleTimeBeforeShutdown = "string"
```
Comment (Nepomuceno) [inline main.tf:88]: there is a shortcut to comment code `del`

**Review by Nepomuceno — APPROVED** [review]
LGTM just small changes.

**Comment by vyta** [comment]
Should this be required for managed vnet in order to enable container builds on the compute instance? Or should the container builds be a separate config option?

File: variables.tf [hunk 2]
```diff
@@ -75,6 +75,12 @@ DESCRIPTION
   }
 }
 
+variable "create_compute_instance" {
+  type        = bool
+  default     = true
```
Comment (vyta) [inline variables.tf:80]: default to false

**Review by vyta — CHANGES_REQUESTED** [review]
I think the compute instance should be a configurable option - created #35 to address this in future changes

File: main.tf (hunk 1 above)
Comment (oloomi) [reply main.tf]: Commented code is now removed.

File: variables.tf (hunk 2 above)
Comment (oloomi) [reply variables.tf]: Good point!  Default value is set to false.

**Comment by oloomi** [comment]
> Should this be required for managed vnet in order to enable container builds on the compute instance? Or should the container builds be a separate config option?

That's right, when the Azure Container Registry for our workspace is behind a virtual network, it can't be used to directly build Docker images. Instead, we need to configure our AML workspace to use a compute cluster or compute instance to build images.
 
Let's use this compute instance as the image builder and it serves both purposes: provisions the managed network, and it can be used for image building.
 
We need to use an Azure CLI command (or a Python code) after the workspace and compute instance are created to update the workspace and set the `image-build-compute` parameter to the created compute instance.

Setting the `imageBuildComput` property in the workspace at this stage creates a cycle and causes Terraform to fail.

**Review by vyta — APPROVED** [review]

**Pull request merged**